File: docs/source/api/user-guide/plugins/code/measurement.py
import opendp.prelude as dp

dp.enable_features("honest-but-curious", "contrib")


# make-base-constant
def make_base_constant(constant):
    """Construct a Measurement that only returns a constant value."""

    def function(_arg: int):
        return constant

    def privacy_map(d_in: int) -> float:
        return 0.0

    return dp.m.make_user_measurement(
        input_domain=dp.atom_domain(T=int),
        input_metric=dp.absolute_distance(T=int),
        output_measure=dp.max_divergence(),
        function=function,
        privacy_map=privacy_map,
        TO=type(constant),
    )


# /make-base-constant


# use-measurement
meas = (
    (
        dp.vector_domain(dp.atom_domain((0, 10))),
        dp.symmetric_distance(),
    )
    >> dp.t.then_sum()
    >> make_base_constant("denied")
)
meas([2, 3, 4])
meas.map(1)
# /use-measurement

File: docs/source/api/user-guide/plugins/code/measurement.R
library(opendp)

enable_features("honest-but-curious", "contrib")


# make-base-constant
make_base_constant <- function(input_domain, input_metric, constant) {
  function_ <- function(.arg) constant
  privacy_map <- function(d_in) 0.

  make_user_measurement(
    input_domain = input_domain,
    input_metric = input_metric,
    output_measure = pure_dp(),
    function_ = function_,
    privacy_map = privacy_map,
    .TO = String
  )
}
then_base_constant <- to_then(make_base_constant)
# /make-base-constant


# use-measurement
space <- c(
  vector_domain(atom_domain(bounds = c(0L, 10L))),
  symmetric_distance()
)
meas <- space |>
  then_sum() |>
  then_base_constant("denied")
meas(arg = c(2L, 3L, 4L))
meas(d_in = 1L)
# /use-measurement
